src/fjaelllada/db_sync.py

```diff
@@ -0,0 +1,141 @@
+import json
+
+from webdav3.client import Client
+from pathlib import Path
+import os
+from datetime import datetime
+import time
+import re
+from filelock import FileLock
+
+# Configuration
+nextcloud_url = "https://cloud.jdav-freiburg.de/remote.php/dav/files/depot"
+api_token = "pwERt-no4Cd-gQQ3A-sQpqc-ZxHz4"
+username = "depot"
+local_file_paths = []
+remote_file_paths = []
+
+# WebDAV client setup
+options = {
+    'webdav_hostname': nextcloud_url,
+    'webdav_login': username,
+    'webdav_password': api_token
+}
+
+client = Client(options)
+
+def get_sync_files_from_json():
+    global local_file_paths
+    global remote_file_paths
+    config_path = os.getenv("config_path")
+    with open(config_path) as json_file:
+        conf = json.load(json_file)
+    local_file_paths = conf["database"]["local_path"]
+    remote_file_paths = conf["database"]["remote_path"]
+
+# Upload a file
+def upload_file(local_path, remote_path):
```

Comment:
Add python typing, use `pathlib.Path` here
```suggestion
def upload_file(local_path: Path, remote_path: str):
```